with-firebase-storage-signed-url-upload/App.jsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import * as ImagePicker from "expo-image-picker";
+import {getApps, initializeApp} from "firebase/app";
+import {getDownloadURL, getStorage, ref} from "firebase/storage";
+import {getFunctions, httpsCallable} from "firebase/functions";
+import {getAuth, signInAnonymously} from "firebase/auth";
+import * as FileSystem from "expo-file-system";
+import * as mime from 'react-native-mime-types';
+
+import {
+  ActivityIndicator,
+  Button,
+  Image,
+  LogBox,
+  Platform,
+  Share,
+  StatusBar,
+  StyleSheet,
+  Text,
+  View,
+} from "react-native";
+import * as Clipboard from "expo-clipboard";
+import uuid from "uuid";
+import {MediaTypeOptions} from "expo-image-picker/src/ImagePicker.types";
+
+const firebaseConfig = {
+  // YOUR FIREBASE CONFIG
+};
+
+// Editing this file with fast refresh will reinitialize the app on every refresh, let's not do that
+if (!getApps().length) {
+  initializeApp(firebaseConfig);
+}
+
+// Firebase sets some timeers for a long period, which will trigger some warnings. Let's turn that off for this example
+LogBox.ignoreLogs([`Setting a timer for a long period`]);
+
+export default class App extends React.Component {
```

Comment:
All new examples should use the latest React functionality ie functional components and hooks.